src/sdam/description/topology/mod.rs

```diff
@@ -85,7 +85,6 @@ impl PartialEq for TopologyDescription {
         self.compatibility_error == other.compatibility_error
             && self.servers == other.servers
             && self.topology_type == other.topology_type
-            && self.cluster_time == other.cluster_time
```

Comment:
This is another unrelated bug; we should not be considering topology descriptions to be different if only the cluster time is different, or else we'll be notifying the server selection operations that they should wake up to try to select a server when nothing that actually matters to them has changed.